starknet_devnet/util.py

```diff
@@ -263,3 +264,62 @@ def __exit__(self, exc_type, exc_val, exc_tb):
 # FeederGatewayClient is implemented in such a way that it logs and raises;
 # this suppresses the logging
 suppress_feeder_gateway_client_logger = LogSuppressor("services.external_api.client")
+
+logger = logging.getLogger("gunicorn.error")
+
+
+class LogContext(dict):
+    """Context manager for logging transactions accumulating info while executing transaction"""
+
+    def get_context_name(self) -> str:
+        """Getting name of current context to put as first line in log entry"""
+        return getattr(self, "name", None) or "Transaction"
+
+    def set_context_name(self, name):
+        """Setting name of current context to put as first line in log entry"""
+        setattr(self, "name", name)
+        return self
+
+    def cust_fees_to_int(self):
+        """Cust to human readable type"""
+        for key in ["actual_fee", "max_fee"]:
+            if self.get(key):
+                try:
+                    self[key] = int(self[key], 16)
+                except (ValueError, TypeError):
+                    pass
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.cust_fees_to_int()
+        if exc_type:
+            if isinstance(exc_val, StarkException):
+                error_message = {
+                    "message": exc_val.message,
+                    "code": str(exc_val.code),
+                    "context": self,
+                }
+            else:
+                error_message = {"message": str(exc_val), "context": self}
+            logger.error(
+                "%s failed: %s", self.get_context_name(), pprint.pformat(error_message)
```

Comment:
What's the benefit of using `pprint`? Asking because I'm not experienced with it.

Reply:
It prints json with indentation (like jq cli tool) and it's very readable